### DISCUSSION TITLE
Disabled buttons in the thumbnail settings

### DISCUSSION THREAD

**Discussion by kopieMK** [pimcore/pimcore#12037] [Q&A / Support] [answered]
Hi. I have updated another box from version 6.9.6 to 10.4.0. 

I can now no longer create thumbnail definitions in the backend. The button "add" and the button "save" are disabled. See screenshot. The save button shows me a tooltip "Not writable. This could be due to several reasons. Please see the documentation for details". I have no idea myself what to look for here. 

I got the hint that I should run this command from the upgrade notes:

``` 
bin/console pimcore:migrate:thumbnails-folder-structure 
```

Unfortunately, this did not change anything. 
Does anyone of you have an idea about this?

Danke!

![thumbnail-settings](https://user-images.githubusercontent.com/50910488/166936315-dab1d59a-748f-4aa3-8f8e-031a54a02c5f.jpg)

**Comment by dvesh3** [comment]
please see https://pimcore.com/docs/pimcore/current/Development_Documentation/Deployment/Configuration_Environments.html#page_Configuration-Storage-Locations-Fallbacks

**Comment by kopieMK** [comment]
Ok, thanks for the link. I'm not sure yet if I can extract the solution there, but it gave me the idea to test the whole thing in the different environments. Dev and staging goes the function. In env=prod it does not. That's already a realization. I'm now trying to understand what it says, because probably there is the solution and I just do not understand it 😂

**Comment by kopieMK** [comment, marked as answer]
Good morning. I think I found the solution that works for me. The Link from @dvesh3 above contains the awnser. In the .env I set ``` PIMCORE_WRITE_TARGET_IMAGE_THUMBNAILS=settings-store ``` . Since this it was possible to add or save a thumbnail config. I hope this helps other with such a problem. Thank you @dvesh3

**Reply by dvesh3** [reply]
yes, if you would like to change configs on the production environment directly, then you have to use `settings-store` otherwise you would only push changes from staging to prod.
